### ISSUE TITLE
Canvas doesn't use loaded fonts as fallback font

### ISSUE THREAD

**Issue by Acituanbus** [Automattic/node-canvas#1997]
## Issue or Feature
Node Canvas doesn't fall back to loaded fonts if it encounters an unknown character

When loading fonts with `canvas.registerFont()`, the loaded fonts aren't used if they support a character the mainly used font does not, ONLY if fonts are installed on the system.
That means, if I run this on a VPS without any installed fonts, it works as expected, falling back to whatever font supports a character if it encounters one the main font doesn't support.
But if I run it on my local machine, this doesn't happen.

I'm faily sure this issue lies with installed system fonts, because a variation of this also occurs when trying to use a font file with the same font family as a system font (see below for more details)

## Steps to Reproduce
```javascript
//load a fallback font that DEFINITELY supports the character (code2000)
canvas.registerFont('./Assets/Fonts/code2000.ttf', {family: 'Code2000'})

//load the Uni Sans Heavy custom font
canvas.registerFont('./Assets/Fonts/Uni_Sans_Heavy.ttf', {family: 'Uni Sans'})

let cnvs = canvas.createCanvas(740, 260)
let ctx = cnvs.getContext('2d')
ctx.fillStyle = "#fff"
ctx.font = '50px Uni Sans'
ctx.fillText(`HELLO WORLD, I'M ᗩcł!`, 50, 100);
    
//result doesn't display the "ᗩ" character correctly on my local machine, but everything works fine on a VPS

Result on local machine: https://i.stack.imgur.com/yR2ue.png


##Using a font file with the same font family as a system font
Like mentioned above, something similar occurs when trying to use a font file, but having a font with the same name installed on the system
```javascript
canvas.registerFont('/Assets/Fonts/uni_sans_heavy.ttf', {family: 'Uni Sans'})

let cnvs = canvas.createCanvas(500, 150)
let ctx = cnvs.getContext('2d')
    
ctx.font = 'bold 50px Uni Sans'
    
ctx.fillText(`This is a test!`, 25, 50)
```
The above code DOESN'T use the Uni_Sans_Heavy.ttf file I register, nor does it use the system font, instead it uses another (unknown to me) font, when run on my local machine. 
However, before I installed the UniSans font to my system, everything worked fine. Running the code on my VPS, which doesn't have any installed fonts either also produces no issues.

**tl;dr: Node Canvas fonts behave strangely with system fonts and font files clashing in certain instances. Is there a way to force it to use a font file instead? Is there a way to force it to use certain fonts as fallback fonts?**

## Your Environment
* Version of node-canvas (output of `npm list canvas` or `yarn list canvas`): 2.9.0
* Environment (e.g. node 4.2.0 on Mac OS X 10.8): Node v16.14.0 on Windows 11

**Comment by chearon** [comment]
Did you mean to write `ctx.font = '50px Uni Sans, Code2000`? The browser wouldn't load your `Code2000` font if you only registered, but never referenced the `Code2000` font either. I think your VPS is the unexpected scenario, whereas your local machine not picking `Code2000` makes sense.

For the second problem with the installed fonts, is that happening on Windows but registering installed fonts does work on Linux?

Platform font APIs don't let us pick fonts by file, so font selection is an approximation and only works most of the time until we rewrite to do our own font selection (big job).

**Comment by Aci-yt** [comment]
Using `ctx.font = '50px Uni Sans, Code2000` ends up with the same result as if I just used `ctx.font = '50px Code2000`, as if Uni Sans didn't exist at all.

And yes, the second problem happened after I installed the "Uni Sans" font on my Windows 11 PC, but my Debian 11 VPS still uses the .ttf file. 
I'm not sure if it comes with pre-installed fonts, but I didn't install any myself.
I'll see if there's a way to check pre-installed fonts on linux so I can check if maybe fonts that support that special character were simply pre-installed on my VPS and not my local machine, which could have caused the difference.

**Comment by Aci-yt** [comment]
I checked, and the only font installed is DejaVuSans..

**Comment by mengsumin** [comment]
is any one fixed this？  i alse happed same things？sometimes it show blank， sometimes it show four number

**Comment by jamesjoung** [comment]
otf and ttf family name must match with family name in the metadata of the font family.

**Comment by teewuane** [comment]
So it will do fallback fonts, but only with system installed fonts, and not with ones I specify? Do I understand this correctly?
